language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/connector/ConnectorNodeVisitor.java

```diff
@@ -0,0 +1,1053 @@
+    private boolean found = false;
+    private List<BLangTypeDefinition> connectors;
+    private Map<String, BLangTypeDefinition> records;
+
+    public ConnectorNodeVisitor(String name) {
+        this.name = name;
+        connectors = new ArrayList<>();
+        records = new HashMap<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<BLangTypeDefinition> getConnectors() {
+        return connectors;
+    }
+
+    public Map<String, BLangTypeDefinition> getRecords() {
+        return records;
+    }
+
+    @Override
+    public void visit(BLangPackage pkgNode) {
+        List<TopLevelNode> topLevelNodes = pkgNode.topLevelNodes;
+        pkgNode.getImports().forEach(importPackage -> {
+            if (importPackage.symbol != null) {
+                this.packageMap.put(importPackage.symbol.pkgID, importPackage);
+            }
+        });
+        topLevelNodes.stream()
+                .filter(CommonUtil.checkInvalidTypesDefs())
+                .forEach(topLevelNode -> ((BLangNode) topLevelNode).accept(this));
+    }
+
+    @Override
+    public void visit(BLangTestablePackage testablePkgNode) {
+        // No implementation
+    }
+
+    @Override
+    public void visit(BLangCompilationUnit compUnit) {
+        if (!found) {
+            compUnit.getTopLevelNodes().forEach(n -> ((BLangNode) n).accept(this));
+        }
+    }
+
+    @Override
+    public void visit(BLangImportPackage importPkgNode) {
+        // No implementation
+    }
+
+    @Override
+    public void visit(BLangXMLNS xmlnsNode) {
+        // No implementation
+    }
+
+    @Override
+    public void visit(BLangFunction funcNode) {
+        // No implementation
+//        if (found) {
+//            if ("__init".equals(funcNode.getName().getValue())) {
+//                addFunction(funcNode, true);
+//            } else if (funcNode.getFlags().contains(Flag.REMOTE) && funcNode.getFlags().contains(Flag.PUBLIC)) {
+//                addFunction(funcNode, false);
+//            }
+//        }
+    }
+
+    @Override
+    public void visit(BLangService serviceNode) {
+        // No implementation
+    }
+
+    @Override
+    public void visit(BLangResource resourceNode) {
+        // No implementation
+    }
+
+    @Override
+    public void visit(BLangTypeDefinition typeDefinition) {
+        if (typeDefinition.getTypeNode() instanceof BLangObjectTypeNode) {
+            if (((BLangObjectTypeNode) typeDefinition.getTypeNode()).flagSet.contains(Flag.CLIENT)) {
+                this.connectors.add(typeDefinition);
+                typeDefinition.getTypeNode().accept(this);
+            }
+
+        } else if (typeDefinition.getTypeNode() instanceof BLangRecordTypeNode) {
+            this.records.put(((BLangRecordTypeNode) typeDefinition.getTypeNode()).symbol.type.toString(),
+                    typeDefinition);
+        }
+    }
+
+    @Override
+    public void visit(BLangConstant constant) {
```

Comment:
We can skip adding these empty methods since already handled same in the parent visitor (BaseNodeVisitor)

Reply:
Fixed in 4c86670d0ff50290929d988e4d6d6fcd0d2f0bdd